iceoryx2-cli/iox2/src/commands.rs

```diff
@@ -286,3 +286,157 @@ where
 pub fn execute(command_name: &str, args: Option<&[String]>) -> Result<()> {
     execute_impl::<HostEnvironment>(command_name, args)
 }
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use iceoryx2_bb_testing::assert_that;
+    use std::env;
+    use std::fs::File;
+    use tempfile::TempDir;
+
+    const IOX2_PREFIX: &str = "iox2-";
+    const FOO_COMMAND: &str = "Xt7bK9pL";
+    const BAR_COMMAND: &str = "m3Qf8RzN";
+    const BAZ_COMMAND: &str = "P5hJ2wAc";
+
+    fn create_noop_executable(file_path: &std::path::Path) -> std::io::Result<()> {
+        use std::process::Command;
+
+        let src_file = file_path.with_extension("rs");
+        std::fs::write(&src_file, "fn main() {}")?;
+        let output = Command::new("rustc")
+            .arg(&src_file)
+            .arg("-o")
+            .arg(file_path)
+            .arg("--crate-type")
+            .arg("bin")
+            .output()?;
```

Comment:
Does this work for cross-compilation?

I was thinking about having a build.rs file to generate the test binara, but I guess it would have the same issue.

Reply:
@elBoberido Perhaps not... I say we tackle it when we want to make the cli no_std.